web_src/css/features/heatmap.css

```diff
@@ -1,8 +1,15 @@
+#user-heatmap-container {
+  min-height: 140px;
+}
+
+.user.profile #user-heatmap-container {
+  min-height: 170px;
+}
```

Comment:
These styles don't seem right. There are more `@media` styles below.

And I think the `#user-heatmap` ID could be kept (without introducing `#user-heatmap-container`), just remove the duplicate ID from the Vue component.

Reply:
Heatmap scaling is weird, this needs more adjustments to correctly guess the height so content won't shift.

Reply:
The heatmap is also fixed now, by removing some code. 294177c78ba83d10f3b71e937f3afbbbb161e2e8

![image](https://github.com/go-gitea/gitea/assets/2114189/45b3a47e-0e86-4ab8-96d7-32bbe0898caf)

![image](https://github.com/go-gitea/gitea/assets/2114189/d1af3fac-d253-43a2-a6e9-743f9a201a02)

Reply:
Ah, nice trick with the `aspect-ratio`, I will test it further.

Reply:
It's still pretty much a hack because the height of the rendered element is not deterministic at all, only https://github.com/razorness/vue3-calendar-heatmap/issues/26 will help.

I added a second ratio for desktop-size user profile heatmap, but it starts to fail a bit on zoom in, still better than before.